client/engine/chainservice/utils/utils.go

```diff
@@ -7,18 +7,33 @@ import (
 	"github.com/ethereum/go-ethereum/accounts/abi/bind"
 	ethcrypto "github.com/ethereum/go-ethereum/crypto"
 	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/rpc"
 )
 
 // ConnectToChain connects to the chain at the given url and returns a client and a transactor.
-func ConnectToChain(ctx context.Context, chainUrl string, chainPK []byte) (*ethclient.Client, *bind.TransactOpts, error) {
-	client, err := ethclient.Dial(chainUrl)
+func ConnectToChain(ctx context.Context, chainUrl, chainAuthToken string, chainPK []byte) (*ethclient.Client, *bind.TransactOpts, error) {
+	var rpcClient *rpc.Client
+	var err error
+
+	if chainAuthToken != "" {
+		fmt.Println("Adding bearer token authorization header to chain service")
```

Comment:
Is there a service-wide logger I should use instead of `fmt.Print`?

Reply:
You could have this function accept a logger (say a suitable interface like `io.Writer`). Then, the calling context can pass in something suitable. 

For example, the chainservice has a logger https://github.com/statechannels/go-nitro/blob/3cb69815a77fe6daac09ac1d02bedb7c4d6b882e/client/engine/chainservice/eth_chainservice.go#L46. The `logDestination` is in scope when constructing the chain service, when `ConnectToChain` is called.